test/parallel/test-buffer.js

```diff
@@ -289,8 +291,25 @@ for (var i = 0; i < asciiString.length; i++) {
   assert.equal(sliceA[i], sliceB[i]);
 }
 
-// TODO utf8 slice tests
+// UTF-8 slice test
+
+var utf8String = '¡hèlló wôrld!';
+var offset = 100;
+
+b.write(utf8String, 0, Buffer.byteLength(utf8String), 'utf8');
+var utf8Slice = b.toString('utf8', 0, Buffer.byteLength(utf8String));
+assert.equal(utf8String, utf8Slice);
 
+var written = b.write(utf8String, offset, 'utf8');
+assert.equal(Buffer.byteLength(utf8String), written);
+utf8Slice = b.toString('utf8', offset, offset + Buffer.byteLength(utf8String));
+assert.equal(utf8String, utf8Slice);
+
+var sliceA = b.slice(offset, offset + Buffer.byteLength(utf8String.length));
```

Comment:
Are you sure the `Buffer.byteLength` call is what you want here? It will coerce the length to a string, then give you the length of that.